org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/internal/GradleDistributionInfo.java

```diff
@@ -18,27 +18,64 @@
 
 import org.eclipse.osgi.util.NLS;
 
-import org.eclipse.buildship.core.internal.GradlePluginsRuntimeException;
+import org.eclipse.buildship.core.FixedVersionGradleDistribution;
+import org.eclipse.buildship.core.GradleDistribution;
+import org.eclipse.buildship.core.GradleDistributions;
+import org.eclipse.buildship.core.LocalGradleDistribution;
+import org.eclipse.buildship.core.RemoteGradleDistribution;
+import org.eclipse.buildship.core.WrapperGradleDistribution;
 import org.eclipse.buildship.core.internal.i18n.CoreMessages;
 import org.eclipse.buildship.core.internal.util.binding.Validator;
 
 /**
- * Describes a valid or invalid {@link GradleDistribution}.
+ * Describes a valid or invalid {@link BaseGradleDistribution}.
  *
  * @author Donat Csikos
  */
 public final class GradleDistributionInfo {
```

Comment:
⭕️ Why do we still have this class, now that the distribution types expose their information?

Reply:
`GradleDistributionInfo` can represent invalid states. It's needed for example when the user configures an invalid URL on the preferences pages. Besides the info object is there to serialize and deserialize distributions.

Reply:
Got it, thanks! A common term for this is `ViewModel`, maybe call it like that?

Reply:
`GradleDistributionViewModel`?

Reply:
Yeah, exactly

Reply:
And it can live in the UI plugin

Reply:
Well, the serialization part is used in the core plugin to persist build configurations.

Reply:
Shouldn't that be part of the `GradleDistribution` class (or BuildConfiguration)?

Reply:
That might make sense. The serialization is used in different configuration (workspace/build/run). Besides, a few dependent classes are in `core` but they should be in `ui`. I'll see what can I do about that.

Reply:
Since we need to keep the serialized form backwards compatible anyway, I think making it part of the API is fine. I.e. have a `toString()` and `fromString()`.

Reply:
I've added `fromString()` and `toString()` to the pubic API, renamed `GradleDistributioninfo` to `GradleDistributionViewModel` and moved it to the `ui` plugin.